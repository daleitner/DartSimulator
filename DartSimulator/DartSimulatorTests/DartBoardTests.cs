using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DartSimulator;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Should;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;

namespace DartSimulatorTests
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class DartBoardTests
	{
		[TestMethod]
		public void DartBoardShouldHaveGotAllFields()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields = dartBoard.Fields.Select(x => x.ToString());
			Approvals.VerifyAll(fields, "Field");
		}

		[TestMethod]
		public void GetDoubleBullShouldReturnDoubleBullField()
		{
			var dartBoard = DartBoard.GetInstance();
			var doubleBull = dartBoard.GetDoubleBull();
			Approvals.Verify(doubleBull);
		}

		[TestMethod]
		public void GetSingleBullShouldReturnSingleBullField()
		{
			var dartBoard = DartBoard.GetInstance();
			var singleBull = dartBoard.GetSingleBull();
			Approvals.Verify(singleBull);
		}

		[TestMethod]
		public void GetSingleInFieldShouldReturnSingleInFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.SingleIn).Select(x => dartBoard.GetSingleInField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetSingleOutFieldShouldReturnSingleOutFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.SingleOut).Select(x => dartBoard.GetSingleOutField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetDoubleFieldShouldReturnDoubleFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.Double).Select(x => dartBoard.GetDoubleField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetTripleFieldShouldReturnTripleFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.Triple).Select(x => dartBoard.GetTripleField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}

		[TestMethod]
		public void VerifyNeighboursOfFields()
		{
			var dartBoard = DartBoard.GetInstance();
			string verifyString = "";
			foreach (var field in dartBoard.Fields)
			{
				verifyString += field.ToString() + "\n";
				foreach (var neighbour in field.Neighbours)
				{
					verifyString += "\t" + neighbour.ToString() + "\n";
				}
			}
			Approvals.Verify(verifyString);
		}
	}
}
