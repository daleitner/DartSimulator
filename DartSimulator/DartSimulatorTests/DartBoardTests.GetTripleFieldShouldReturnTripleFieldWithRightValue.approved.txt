Field[0] = T1
Field[1] = T2
Field[2] = T3
Field[3] = T4
Field[4] = T5
Field[5] = T6
Field[6] = T7
Field[7] = T8
Field[8] = T9
Field[9] = T10
Field[10] = T11
Field[11] = T12
Field[12] = T13
Field[13] = T14
Field[14] = T15
Field[15] = T16
Field[16] = T17
Field[17] = T18
Field[18] = T19
Field[19] = T20
